app_legacy/Helpers/database/user-account-deletion.php

```diff
@@ -80,6 +81,20 @@ function deleteOverdueUserAccounts(): void
 
 function clearAccountData(User $user): void
 {
+    // disable account access while we destroy it (prevents creating new records during delete)
+    legacyDbStatement("UPDATE UserAccounts u SET
+        u.Password = null,
+        u.SaltedPass = '',
+        u.cookie = null,
+        u.appToken = null,
+        u.APIKey = null
+        WHERE u.ID = :userId", ['userId' => $user->ID]
+    );
+
+    // reset all achievements earned by the player
+    $action = new ResetPlayerAchievementAction();
+    $action->execute($user);
```

Comment:
This ensures contribution yields are updated correctly for any achievements the deleted user had earned.

It also updates their score to 0 and deletes any game-related badges they may have had, which are redundantly handled later.